### PR TITLE
Fix dependencies spec and add CI testing with -Zminimal-versions

### PR DESCRIPTION
**Pull request by asomers**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -17,15 +17,15 @@ travis-ci = { repository = "eqrion/cbindgen" }
 [dependencies]
 clap = { version = "2", optional = true }
 log = "0.4"
-serde = { version = "1.0", default-features = false, features = ["derive"]}
+serde = { version = "1.0.103", default-features = false, features = ["derive"]}
```
Comment (emilio) [inline Cargo.toml:20]: Can you elaborate? Why do we need this change?

Comment (asomers) [reply Cargo.toml:20]: As-is, the dependencies are incorrectly specified.  If you do `cargo +nightly update -Zminimal-versions && cargo +nightly build`, the build will fail.  Worse, any crate that depends on cbindgen will also fail to build if it tries building with its minimal dependencies.

**Comment by emilio** [comment]
Thanks!

**Comment by emilio** [comment]
https://travis-ci.org/github/eqrion/cbindgen/builds/672741687 is green (not sure what's up with GitHub's integration).

**Pull request merged**